tracing-forest/src/tree/mod.rs

```diff
@@ -244,9 +256,26 @@ impl Span {
             total_duration: Duration::ZERO,
             inner_duration: Duration::ZERO,
             nodes: Vec::new(),
+            defer_unless_children_attached: false,
         }
     }
 
+    #[cfg(feature = "defer")]
+    pub(crate) fn defer_unless_children_attached(mut self, defer: bool) -> Self {
+        self.defer_unless_children_attached = defer;
+        self
+    }
+
+    #[cfg(feature = "defer")]
+    pub(crate) fn should_render(&self) -> bool {
+        !(self.defer_unless_children_attached && self.nodes().is_empty())
+    }
+
+    #[cfg(not(feature = "defer"))]
+    pub(crate) fn should_render(&self) -> bool {
```

Comment:
nit: unify with the above to just one function

Reply:
You can't, not with the cfg features. It gets really messy if you unify it.